src/ripple/app/misc/impl/ValidatorList.cpp

```diff
@@ -195,29 +206,88 @@ ValidatorList::load(
 }
 
 boost::filesystem::path
-ValidatorList::GetCacheFileName(PublicKey const& pubKey)
+ValidatorList::getCacheFileName(PublicKey const& pubKey) const
 {
     return dataPath_ / (filePrefix_ + strHex(pubKey));
 }
 
+// static
+Json::Value
+ValidatorList::buildFileData(
+    std::string const& pubKey,
+    ValidatorList::PublisherListCollection const& pubCollection,
+    boost::optional<std::uint32_t> forceVersion /* = {} */)
+{
+    Json::Value value(Json::objectValue);
+
+    auto const effectiveVersion = forceVersion
+        ? *forceVersion
+        : (pubCollection.remaining.empty() ? pubCollection.rawVersion : 2);
```

Comment:
Is there a reason for not simply using `pubCollection.rawVersion` whenever `forceVersion` is unseated? It seems a bit simpler and I'm not sure of why we'd opt for the downgraded format since the goal is to eventually migrate to version 2.

Reply:
Actually, there is one reason, but it's extremely unlikely. It's possible for a peer to send a version 1 protocol message which contains a VL with a future effective date. That would cause `rawVersion` to be 1, even though we _could_ be building a version 2 value.

What I'm going to do is change the code here to always use `rawVersion`, but also include an assert to make sure `rawVersion` is "correct". I'm also going to change the `rawVersion` assignment so that if we get into this situation, it forces the version to 2.